yaks/lib/yaks/format/json_api.rb

```diff
@@ -42,31 +42,31 @@ def serialize_resource(resource)
         attributes = resource.attributes.reject { |k| k.equal?(:id) }
         result[:attributes] = attributes if attributes.any?
 
-        result[:links] = {}
-        result[:links].update(serialize_subresource_links(resource.subresources))
-        result[:links].update(serialize_links(resource.links))
-        result.delete(:links) if result[:links].empty?
+        relationships = serialize_relationships(resource.subresources)
+        result[:relationships] = relationships unless relationships.empty?
+        links = serialize_links(resource.links)
+        result[:links] = links unless links.empty?
 
         result
       end
 
-      # @param [Yaks::Resource] subresource
+      # @param [Array] subresources
       # @return [Hash]
-      def serialize_subresource_links(subresources)
+      def serialize_relationships(subresources)
         subresources.each_with_object({}) do |resource, hsh|
-          next if resource.null_resource?
-          hsh[resource.rels.first.sub(/^rel:/, '')] = serialize_subresource_link(resource)
+          hsh[resource.rels.first.sub(/^rel:/, '').to_sym] = serialize_relationship(resource)
```

Comment:
I know this regexp substitution was already there, but it's too naïve. At a minimum it should match against the configured `rel_template`. However rels need to be reïfied anyway (represented as objects instead of symbols/strings), in which case this will become something like. This will be an important step also for implementing #19 (JSON-LD) and possibly #9 (CURIEs).

```
resource.rels.first.short_name
```

So I'm fine with leaving this for now, I'll cook something up to show where I  want to go with rels.